com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/archive/ManifestBuilder.java

```diff
@@ -492,23 +492,19 @@ public ManifestData buildManifestData() throws IOException {
 
         builder.addAllEngineVersions(this.supportedEngineVersions);
 
-        int resourceIndex = 0;
-        HashMap<String, Integer> resourceToIndex = new HashMap<>(); // at what index is the resource stored?
-        for (ResourceEntry entry : this.resourceEntries) {
-            resourceToIndex.put(entry.getUrl(), resourceIndex);
-            resourceIndex++;
-        }
-
         for (ResourceEntry entry : this.resourceEntries) {
             String url = entry.getUrl();
             ResourceEntry.Builder resourceEntryBuilder = entry.toBuilder();
 
-            // Since we'll only ever ask collection proxies, we only store those lists
-            ResourceNode proxyNode = resourceGraph.getResourceNodeFromPath(url);
-            // We'll only store the dependencies for the excluded collection proxies
-            if (proxyNode != null && proxyNode.checkType(ResourceNode.Type.ExcludedCollectionProxy)) {
-                HashSet<ResourceNode> allProxyDependants = this.getAllDependants(proxyNode);
-                for (ResourceNode dependant : allProxyDependants) {
+            // Since we'll only ever ask collection proxies we store lists only for
+            // collections in such proxies. We can't store it in collection proxy itself
+            // because it is possible to change collection for a collection proxy
```

Comment:
Taking into account that now we can change collection which will be loaded by an excluded collection proxy, we have to collect dependencies for collection in the excluded proxy, not for the proxy itself. This way `colelctionproxy.get_resources()` and `colelctionproxy.missing_resources()` will work even after the collection in a collection proxy has been replaced to another one